cmd/koord-descheduler/app/server.go

```
@@ -236,7 +240,18 @@ func buildHandlerChain(handler http.Handler) http.Handler {
 
 func installMetricHandler(pathRecorderMux *mux.PathRecorderMux) {
 	configz.InstallHandler(pathRecorderMux)
-	pathRecorderMux.Handle("/metrics", legacyregistry.HandlerWithReset())
+	pathRecorderMux.Handle("/metrics", promhttp.HandlerFor(MergedRegistry{prometheus.Gatherers{legacyregistry.DefaultGatherer, ctrlmetrics.Registry}}, promhttp.HandlerOpts{
+		ErrorHandling: promhttp.HTTPErrorOnError,
+	}))
+}
+
+type MergedRegistry struct {
+	prometheus.Gatherers
+}
+
+func (mr MergedRegistry) Gather() ([]*dto.MetricFamily, error) {
+	mfs, _ := mr.Gatherers.Gather()
```

we should not ignore the error.